tarfs.go

```diff
@@ -40,6 +40,15 @@ type FileSystem struct {
 // a non-empty string, the resulting FileSystem will first look for resources in
 // the local file system before returning an embedded resource.
 func New(tarData []byte, local string) (*FileSystem, error) {
+	return NewWithListener(tarData, local, func(name string, file []byte) []byte {
+		return file
+	})
+}
+
+// NewWithListener creates a new FileSystem using the given in-memory tar data. If local is
```

Comment:
I would rename 'listener' to `transformer` or something like that. Also, the comment could use a description of what this thing is and how it's used.

Reply:
OK just named it "Filter" and added a type and comments for a little more clarity..